### PR TITLE
srb2kart: 1.3 -> 1.6

### PR DESCRIPTION
**Pull request by donovanglover**
Based on #187320

###### Description of changes

https://github.com/STJr/Kart-Public/releases

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/srb2kart/default.nix [hunk 1]
```diff
@@ -50,39 +73,46 @@ in stdenv.mkDerivation rec {
...
 
-  patches = [
-    ./wadlocation.patch
+  # Desktop icon
+  desktopItems = [
+    (makeDesktopItem rec {
+      name = "Sonic Robo Blast 2 Kart";
+      exec = finalAttrs.pname;
+      icon = finalAttrs.pname;
+      comment = finalAttrs.meta.description;
```
Comment (SuperSandro2000) [inline pkgs/games/srb2kart/default.nix:88]:
```suggestion
      comment = meta.description;
```
updating meta should not trigger rebuilds

File: pkgs/games/srb2kart/default.nix [hunk 2]
```diff
@@ -50,39 +73,46 @@ in stdenv.mkDerivation rec {
...
-  preConfigure = ''
-    mkdir assets/installer
-    pushd assets/installer
-    unzip ${installer} "*.kart" srb2.srb
-    popd
-  '';
+    mkdir -p $out/bin $out/share/applications $out/share/pixmaps $out/share/icons
+
+    copyDesktopItems
+
```
Comment (SuperSandro2000) [inline pkgs/games/srb2kart/default.nix:101]:
```suggestion
```
calling this manually is not required

File: pkgs/games/srb2kart/default.nix [hunk 3]
```diff
@@ -50,39 +73,46 @@ in stdenv.mkDerivation rec {
...
+    cp ../srb2.png $out/share/pixmaps/srb2kart.png
+    cp ../srb2.png $out/share/icons/srb2kart.png
+
+    cp bin/srb2kart $out/bin/srb2kart
+    wrapProgram $out/bin/srb2kart --set SRB2WADDIR "${finalAttrs.assets}/share/srb2kart"
```
Comment (SuperSandro2000) [inline pkgs/games/srb2kart/default.nix:106]:
```suggestion
    wrapProgram $out/bin/srb2kart \
      --set SRB2WADDIR "${finalAttrs.assets}/share/srb2kart"
```

File: pkgs/games/srb2kart/default.nix (hunk 2 above)
Comment (donovanglover) [reply pkgs/games/srb2kart/default.nix:101]: Fixed thanks

File: pkgs/games/srb2kart/default.nix (hunk 3 above)
Comment (donovanglover) [reply pkgs/games/srb2kart/default.nix:106]: Changed

File: pkgs/games/srb2kart/default.nix (hunk 1 above)
Comment (donovanglover) [reply pkgs/games/srb2kart/default.nix:88]: Can you explain? I can't remove `finalAttrs` since stdenv doesn't use `rec`

Comment (SuperSandro2000) [reply pkgs/games/srb2kart/default.nix:88]:
```suggestion
      comment = "kart racing mod based on Sonic Robo Blast 2";
```
kinda duplicatd with meta.description, maybe we find something shorter or more fitting for the desktop icon but so we avoid triggering rebuilds when meta is changed

Comment (donovanglover) [reply pkgs/games/srb2kart/default.nix:88]: Oh I understand now, thanks. I think that comment is good and abbreviated the last part to SRB2.

**Comment by SuperSandro2000** [comment]
Can you please rebase?

**Comment by pinkcreeper100** [comment]
@donovanglover #234734 has been merged

**Pull request closed without merging**